app/services/update_wikidata_stats.rb

```diff
@@ -1,32 +1,132 @@
 # frozen_string_literal: true
-
 require_dependency "#{Rails.root}/lib/wikidata_summary_parser"
 require_dependency "#{Rails.root}/lib/importers/wikidata_summary_importer"
+# require the installed wikidata-diff-analyzer gem
+require 'wikidata-diff-analyzer'
 
 class UpdateWikidataStats
+  STATS_CLASSIFICATION = [
+    'claims created', 'claims removed', 'claims changed',
+    'references added', 'references removed', 'references changed',
+    'qualifiers added', 'qualifiers removed', 'qualifiers changed',
+    'aliases added', 'aliases removed', 'aliases changed',
+    'labels added', 'labels removed', 'labels changed',
+    'descriptions added', 'descriptions removed', 'descriptions changed',
+    'interwiki links added', 'interwiki links removed', 'interwiki links updated',
+    'merged to', 'merged from', 'redirects created',
+    'reverts performed', 'restorations performed', 'items cleared',
+    'items created', 'lemmas added', 'lemmas removed', 'lemmas changed',
+    'forms added', 'forms removed', 'forms changed',
+    'senses added', 'senses removed', 'senses changed',
+    'properties created', 'lexeme items created',
+    'representations added', 'representations removed', 'representations changed',
+    'glosses added', 'glosses removed', 'glosses changed',
+    'form claims added', 'form claims removed', 'form claims changed',
+    'sense claims added', 'sense claims removed', 'sense claims changed'
+  ].freeze
+
   def initialize(course)
     @course = course
-    import_summaries
+    # import_summaries
+    update_summary_with_stats
     update_wikidata_statistics
   end
 
   private
 
+  # When summary is nil, instead of fetching edit summaries for each revision, I want to use
+  # wikidata-diff-analyzer to get the stats saved in the database after serializing.
+  # That means while UpdateWikidataStats will be called for a course, all the revisions which
+  # had edit summaries in the summary field be processed with WikidataSummaryParser
+  # if summary is nil, then the stats would be saved in the summary
+
+  def update_summary_with_stats
+    return if wikidata_revisions_without_summaries.empty?
+
+    revision_ids = wikidata_revisions_without_summaries.pluck(:mw_rev_id)
+    analyzed_revisions = WikidataDiffAnalyzer.analyze(revision_ids)[:diffs]
+
+    revision_ids.each do |rev_id|
+      individual_stat = analyzed_revisions[rev_id]
+      # Serialize the individual_stat to JSON format
+      serialized_stat = individual_stat.to_json
+
+      # Find the revision object by mw_rev_id
+      revision = course_revisions.find_by(mw_rev_id: rev_id)
```

Comment:
I think doing a 'find' and 'update' on each revision may be inefficient. Since we've already loaded all the revisions in `wikidata_revisions_without_summaries`, and since the `analyzed_revisions` hash has the IDs as keys already, I think it makes more sense to loop through `wikidata_revisions_without_summaries` and look up the relevant data for each Revision in `analyzed_revisions` via the mw_rev_id. That would cut out the `find_by` database query, and if you wrap the whole loop in `Revision.transaction` (see other examples of `.transaction` in this codebase), it will do all the updates as one database transaction.